tool/src/docs_util.rs

```diff
@@ -0,0 +1,77 @@
+use std::fmt;
+
+use diplomat_core::ast;
+use pulldown_cmark::{BrokenLink, CowStr, Event, LinkType, Options, Parser, Tag};
+
+pub fn markdown_to_rst<W: fmt::Write>(
+    out: &mut W,
+    markdown: &str,
+    write_reference: &dyn Fn(&ast::Path, &mut W) -> fmt::Result,
+) -> fmt::Result {
+    let mut broken_link_callback = |broken: BrokenLink| {
+        Some((
+            CowStr::from(broken.reference.to_string()),
```

Comment:
I feel like we should just leave these broken, yes? Unless we're implementing intra doc links, which is a bunch of work

Reply:
We need this to support inter-doc links, since otherwise we don't get events for the link.

Reply:
Right, but we don't have that support yet, so is this doing anything? (I might me missing something?)

Reply:
We do! You can see us set `in_shortcut` to true when we see the broken link and then generate the reference when we get an `Event::Code`.

Reply:
(anyway this is non blocking)

Reply:
But we should link to an issue about intradoc

Reply:
Wait, hmm, not sure I understand. We _do_ support intra-doc links for types.

Reply:
_oh_, I see now.